frontend/src/components/NewIssueModal/NewIssueModal.tsx

```diff
@@ -107,6 +111,25 @@ const NewIssueModal: React.FC<React.PropsWithChildren<NewIssueModalProps>> = ({
 		],
 	})
 
+	const [mode, setMode] = React.useState('Create Issue')
```

Comment:
Not necessary now, but maybe we can make this a little more type strong with an enum?
```
type mode = 'create_issue' | 'link_issue'
```

Also like using snake or camel case to avoid thinking this is a string we want to use in the UI

Reply:
I've improved type safety with introduction of some type aliases and enum's check them out.
Open to naming suggestions though.